src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/EC/CurveDef.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography.EcDsa.Tests
+{
+    public class CurveDef
+    {
+        public CurveDef() { }
+        public ECCurve Curve;
+        public ECCurve.ECCurveType CurveType;
+        public int KeySize;
+        public bool IncludePrivate;
+        public bool RequiredOnPlatform;
+
+        public bool IsCurveValidOnPlatform
+        {
+            get
+            {
+                // Assume curve is valid if required; tests will fail if not present
+                return RequiredOnPlatform || ECDsaFactory.IsCurveValid(Curve.Oid);
+            }
+        }
+
+        public bool IsCurveTypeEqual(ECCurve.ECCurveType actual)
+        {
+            if (CurveType == actual)
+                return true;
+
+            // Montgomery curves can be expressed as Weierstrasss, so use whichever the platform says it is for the given curve
+            if (CurveType == ECCurve.ECCurveType.PrimeMontgomery && actual == ECCurve.ECCurveType.PrimeShortWeierstrass)
```

Comment:
Do you have an example where you needed this?

To quote Wikipedia: "Any elliptic curve can be written in Weierstrass form." (https://en.wikipedia.org/wiki/Montgomery_curve#Equivalence_with_Weierstrass_curves).  (Note that it also shows the Montgomery <-> Twisted Edwards coefficient conversions)

So maybe all this is trying to do is isolate Prime v Char-2?

Also, Nit: Too many 's's in Weierstrass in the comment :)